tests/cpp/test_multi_matmul_scheduler.cpp

```diff
@@ -0,0 +1,625 @@
+      const std::vector<IterDomain*>& alloc_orig =
+          tv_orig->getAllocationDomain();
+      const std::vector<IterDomain*>& alloc_new = tv_new->getAllocationDomain();
+      ASSERT_EQ(alloc_new.size(), alloc_orig.size()) << suffix;
+      for (size_t i : c10::irange(alloc_orig.size())) {
+        compareIDs(alloc_orig[i], alloc_new[i]);
+        // Print transforms after failure recursing for context
+        if (testing::Test::HasFailure()) {
+          std::cout << "ORIG: " << tv_orig->toString() << std::endl;
+          tv_orig->printTransforms();
+          std::cout << "NEW: " << tv_new->toString() << std::endl;
+          tv_new->printTransforms();
+        }
+        ASSERT_FALSE(testing::Test::HasFailure()) << suffix;
+      }
+    }
+
+    // Inspect definition. If it's a LoadStoreOp, check that the op type and
+    // cache_op match
+    EXPECT_EQ(
+        tv_new->definition() == nullptr, tv_orig->definition() == nullptr);
+    if (auto* lsop_orig = dynamic_cast<LoadStoreOp*>(tv_orig->definition())) {
+      auto lsop_new = dynamic_cast<LoadStoreOp*>(tv_new->definition());
+      ASSERT_TRUE(lsop_new != nullptr) << suffix;
+      EXPECT_EQ(lsop_new->opType(), lsop_orig->opType());
+      EXPECT_EQ(lsop_new->cacheOp(), lsop_orig->cacheOp());
+    } else if (
+        auto* rop_orig = dynamic_cast<ReductionOp*>(tv_orig->definition())) {
+      auto rop_new = dynamic_cast<ReductionOp*>(tv_new->definition());
+      EXPECT_EQ(
+          rop_new->serialGridReductionRequested(),
+          rop_orig->serialGridReductionRequested());
+    }
+
+    // Print suffix whenever a test fails before recursing
+    ASSERT_FALSE(testing::Test::HasFailure()) << suffix;
+
+    Expr* def_orig = tv_orig->definition();
+    Expr* def_new = tv_new->definition();
+    EXPECT_TRUE(def_orig->sameOp(def_new))
+        << "def_orig\n  " << def_orig->toString()
+        << "is not the same as def_new\n  " << def_new->toString() << suffix;
+    EXPECT_EQ(def_orig->attributes().size(), def_new->attributes().size())
+        << suffix;
+    EXPECT_EQ(def_orig->inputs().size(), def_new->inputs().size()) << suffix;
+    for (size_t i : c10::irange(def_orig->inputs().size())) {
+      if (i >= def_new->inputs().size()) {
+        break;
+      }
+      auto* tv_inp_orig = dynamic_cast<TensorView*>(def_orig->input(i));
+      auto* tv_inp_new = dynamic_cast<TensorView*>(def_new->input(i));
+      EXPECT_EQ(tv_inp_orig != nullptr, tv_inp_new != nullptr) << suffix;
+
+      if (!testing::Test::HasFailure() && tv_inp_orig != nullptr &&
+          !tv_inp_orig->isFusionInput()) {
+        // Skip comparing input TVs since their schedules don't affect the
+        // generated kernel
+        EXPECT_FALSE(tv_inp_new->isFusionInput());
+        compareTVs(tv_inp_orig, tv_inp_new);
+      }
+    }
+
+    // Print suffix whenever a test fails in recursive call
+    ASSERT_FALSE(testing::Test::HasFailure()) << suffix;
+  }
+
+  void compareSchedules() {
```

Comment:
What is the future plan of this check? I don't think in the long term we want to keep the legacy scheduler just for testing purposes. Will we completely removing those tests?

Reply:
Yes. Once we remove the old scheduler we will remove this file.